pkg/patch/patch_test.go

```diff
@@ -34,15 +66,25 @@ func TestGeneratePatch(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if !strings.Contains(string(patch), "aws-appmesh-proxy-route-manager:latest") {
+	verifyPatch(t, string(patch), meta)
+}
+
+func verifyPatch(t *testing.T, patch string, meta Meta) {
+	if !strings.Contains(patch, "aws-appmesh-proxy-route-manager:latest") {
 		t.Errorf("Init container image not found")
 	}
 
-	if !strings.Contains(string(patch), meta.Sidecar.ContainerImage) {
+	if !strings.Contains(patch, meta.Sidecar.ContainerImage) {
 		t.Errorf("Sidecar container image not found")
 	}
 
-	if !strings.Contains(string(patch), "amazon/aws-xray-daemon") {
-		t.Errorf("Patch [%s] has no xray", string(patch))
+	if meta.Sidecar.EnableXray {
+		if !strings.Contains(patch, "amazon/aws-xray-daemon") {
+			t.Errorf("No x-ray found")
+		}
+	} else {
+		if strings.Contains(patch, "amazon/aws-xray-daemon") {
+			t.Errorf("X-Ray container found when EnableXray=false")
+		}
 	}
 }
```

Comment:
Thanks for adding the test!

I'll send a follow up PR to use [json-patch](https://github.com/evanphx/json-patch) to validate json patch logic. 

```
        original := []byte(`{"name": "John", "age": 24, "height": 3.21}`)
	patchJSON := []byte(`[
		{"op": "replace", "path": "/name", "value": "Jane"},
		{"op": "remove", "path": "/height"}
	]`)

	patch, err := jsonpatch.DecodePatch(patchJSON)

	patched, err := patch.Apply(original)

```

Reply:
yes that will be useful, thanks